corgi/core/models.py

```diff
@@ -1986,6 +1999,54 @@ def get_upstreams_purls(self, using: str = "read_only") -> set[str]:
         """Return only the purls from the set of all upstream nodes"""
         return set(node.purl for node in self.get_upstreams_nodes(using=using))
 
+    def disassociate_with_product(self, product_ref: ProductModel):
+        """Disassociate this component with the passed in ProductModel and any child ProductModels
+         in that product's hierarchy. This is the reverse of what happens in save_product_taxonomy
+        and is used when a product is removed from product_definitions, or moved to a new place
+        in the product hierarchy, which also requires re-processing those products from scratch
+        Removing proudctvariants is a no-op where a stream does not have errata_info. It's easier to
+        just remove productvariants everytime then check for errata_info in a stream.
+        """
+        if isinstance(product_ref, Product):
+            self.productvariants.remove(*product_ref.productvariants.get_queryset())
+            self.productstreams.remove(*product_ref.productstreams.get_queryset())
+            self.productversions.remove(*product_ref.productversions.get_queryset())
+            self.products.remove(product_ref)
+        elif isinstance(product_ref, ProductVersion):
+            self.productvariants.remove(*product_ref.productvariants.get_queryset())
+            self.productstreams.remove(*product_ref.productstreams.get_queryset())
+            self.productversions.remove(product_ref)
+            self._check_and_remove_orphaned_product_refs(product_ref, "Product")
+        elif isinstance(product_ref, ProductStream):
+            self.productvariants.remove(*product_ref.productvariants.get_queryset())
+            self.productstreams.remove(product_ref)
+            self._check_and_remove_orphaned_product_refs(product_ref, "ProductVersion")
+            self._check_and_remove_orphaned_product_refs(product_ref, "Product")
+        elif isinstance(product_ref, ProductVariant):
+            self.productvariants.remove(product_ref.productvariants)
```

Comment:
Should this just be `product_ref`, or `*product_ref.productvariants.get_queryset()`?
```suggestion
            self.productvariants.remove(product_ref)
```

Reply:
You are correct, it should be product_ref.